R/kartverket.R

```diff
@@ -0,0 +1,211 @@
+#' point.
+#'
+#' @details Advanced users might want to look into the additional possible
+#' parameters. For a list of all accepted query parameters see
+#' [create_params_kv_punktsok()] or refer to the [API
+#' documentation](https://ws.geonorge.no/adresser/v1/#/default/get_sok) for
+#' further details (in Norwegian only).
+#'
+#' The function returns a data.frame or list by default. If you set
+#' `raw_response` to `TRUE`, the raw response from the API will be returned
+#' instead. Note that the response will then be returned silently.
+#'
+#' @param x A vector or list with latitude and longitude coordinates.
+#' @param radius An integer value with the radius for the query, in whole
+#'   meters.
+#' @param crs An integer or vector with the input and output coordinate
+#'   reference system. Default value is ETRS 89 for latitude and longitude data
+#'   (EPSG:4258).
+#' @inheritParams get_address_info
+#' @inheritParams get_entity
+#'
+#' @return data.frame or list
+#' @export
+#' @examples
+#' # Using a vector
+#' df <- find_address_from_point(c(lat = 63.42805, lon = 10.39679))
+#'
+#' # Using a list
+#' dl <- find_address_from_point(
+#'   list(c(lat = 59.91364, lon = 10.7508),
+#'        c(lat = 63.42805, lon = 10.39679)))
+#'
+#' # Using a data.frame
+#' dl <- find_address_from_point(
+#'     data.frame(lat = c(59.91364, 63.42805),
+#'                lon = c(10.7508, 10.39679)))
+#'
+find_address_from_point <- function(x, radius = 50, crs = 4258, ...,
+                                    raw_response = FALSE) {
+  UseMethod('find_address_from_point')
+}
+
+#' find_address_from_point (numeric)
+#' @inheritParams find_address_from_point
+#' @export
+find_address_from_point.numeric <- function(x, radius = 50, crs = 4258, ...,
+                                            raw_response = FALSE) {
+  res <- find_address_from_point_safe(x, radius = radius, crs = crs,
+                                      raw_response = raw_response)
+  if (raw_response) return(invisible(res))
+  res
+}
+
+#' find_address_from_point (list)
+#' @inheritParams find_address_from_point
+#' @export
+find_address_from_point.list <- function(x, radius = 50, crs = 4258, ...,
+                                         raw_response = FALSE) {
+  dl <- lapply(x, find_address_from_point.numeric, radius = radius, crs = crs,
+               ..., raw_response = raw_response)
+  if (raw_response) return(invisible(dl))
+  if (length(dl) == 1) dl <- dl[[1]]
+  dl <- dl[lengths(dl) != 0]
+  dl
+}
+
+#' find_address_from_point (data.frame)
+#' @inheritParams find_address_from_point
+#' @export
+find_address_from_point.data.frame <- function(x, radius = 50, crs = 4258, ...){
+
+  if (!all(c('lat', 'lon') %in% colnames(x)))
+    cli::cli_abort('data.frame must have columns with names lat and lon')
+
+  lapply(1:nrow(x), function(i) {
+    coords <- c(x[i, 'lat'], x[i, 'lon'])
+    find_address_from_point_safe(coords, radius = radius, crs = crs)
+  })
+}
```

Comment:
Good point. We should think closer about this one.

Reply:
This has been changed so that `find_address_from_point.data.frame` now returns a data.frame. Additionally I have added `punkt` as list column to the output from `find_address_from_point_single()`. This  ensures that the method is equal across all methods. 

I'm however a bit sceptical of combing the full input data.frame with the result from the API. I fully get the use case here, but this would mean that the `data.frame` method would return a different result than the other methods for the same function and would also stand out in terms of how functions for the other APIs behave.

Reply:
As discussed I have removed the `data.frame` method for `find_address_from_point()`. I will add an issue for the possibility of supporting enrichment of input data.frames.